internal/setup/setup.go

```diff
@@ -917,3 +945,22 @@ func packageToYAML(p *Package) (*yamlPackage, error) {
 	}
 	return pkg, nil
 }
+
+type Semaphore struct {
```

Comment:
[Comment for reviewer]: This type mostly exists for documentation because the pattern of using a channel as a semaphore can be difficult to understand when reading the code if you have not seen it before.